x-pack/legacy/plugins/reporting/types.d.ts

```diff
@@ -193,9 +198,10 @@ export interface JobParamPostPayload {
 
 export interface JobDocPayload {
   headers?: Record<string, string>;
-  jobParams: object;
+  jobParams: any;
```

Comment:
The JobDocPayload's jobParams depends on what kind of job it is. It could be an object with savedObjectId (CSV export) or a URL (PDF). Getting to a point where there is a "base" `JobDocPayload` type that is extended for more specific contexts, has been hard to keep straight with the way I've tried. For now, I'm switching this to `any` but I think the correct way to do this is with TypeScript Generics.

At some point in the future, I could see that the way to get the `JobDocPayload` type in a specific context would be to use a generic:

```
let fooPayloadA: JobDocPayload<JobParamsSavedObject>;
let fooPayloadB: JobDocPayload<JobParamsUrl>;
```